[2.0.0]  warning: ignoring return value of ‘read’ declared with attribute ‘warn_unused_result’

Just a minor warning

while preparing the Debina package for Debian experimental, I got this

```
src/silx/io/specfile/src/sfinit.c: In function ‘sfAssignScanNumbers’:
src/silx/io/specfile/src/sfinit.c:808:9: warning: ignoring return value of ‘read’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  808 |         read(sf->fd,buffer,sizeof(buffer));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Here a few other :))

```
In file included from /usr/lib/python3/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1940,
                 from /usr/lib/python3/dist-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                 from /usr/lib/python3/dist-packages/numpy/core/include/numpy/arrayobject.h:5,
                 from src/silx/math/marchingcubes.cpp:1230:
/usr/lib/python3/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
      |  ^~~~~~~
src/silx/math/marchingcubes.cpp: In function ‘int __pyx_pymod_exec_marchingcubes(PyObject*)’:
src/silx/math/marchingcubes.cpp:25238:5: warning: converting to non-pointer type ‘int’ from NULL [-Wconversion-null]
25238 |     import_array();
      |     ^~~~~~~~~~~~
src/silx/math/marchingcubes.cpp:25247:5: warning: converting to non-pointer type ‘int’ from NULL [-Wconversion-null]
25247 |     import_umath();
      |     ^~~~~~~~~~~~
```

I do not know if thi is an "issue" with cython :)

--- a/src/silx/io/specfile/src/sfinit.c
+++ b/src/silx/io/specfile/src/sfinit.c
@@ -792,6 +792,7 @@ static void
 sfAssignScanNumbers(SpecFile *sf) {
 
   int i;
+  long bytesread;
   char *ptr;
   char buffer[50];
   char buffer2[50];
@@ -805,7 +806,10 @@ sfAssignScanNumbers(SpecFile *sf) {
         scan = (SpecScan *) object->contents;
 
         lseek(sf->fd,scan->offset,SEEK_SET);
-        read(sf->fd,buffer,sizeof(buffer));
+        bytesread = read(sf->fd,buffer,sizeof(buffer));
+        if (bytesread <= 4) {
+            continue;
+        }
         buffer[49] = '\0';
 
         for ( ptr = buffer+3,i=0; *ptr != ' ';ptr++,i++) buffer2[i] = *ptr;

--- a/src/silx/math/marchingcubes.pyx
+++ b/src/silx/math/marchingcubes.pyx
@@ -39,19 +39,6 @@ cimport cython
 cimport silx.math.mc as mc
 
 
-# From numpy_common.pxi to avoid warnings while compiling C code
-# See this thread:
-# https://mail.python.org/pipermail//cython-devel/2012-March/002137.html
-cdef extern from *:
-    bint FALSE "0"
-    void import_array()
-    void import_umath()
-
-if FALSE:
-    import_array()
-    import_umath()
-
-
 cdef class MarchingCubes:
     """Compute isosurface using marching cubes algorithm.
 

